app/controllers/TraderAgentRequiredInformationController.scala

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import javax.inject.Inject
+
+import scala.concurrent.{ExecutionContext, Future}
+
+import play.api.i18n.{I18nSupport, MessagesApi}
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendBaseController
+
+import controllers.actions._
+import forms.TraderAgentRequiredInformationFormProvider
+import models.{DraftId, NormalMode}
+import navigation.Navigator
+import pages.TraderAgentRequiredInformationPage
+import services.UserAnswersService
+import views.html.TraderAgentRequiredInformationView
+
+class TraderAgentRequiredInformationController @Inject() (
+  override val messagesApi: MessagesApi,
+  userAnswersService: UserAnswersService,
+  navigator: Navigator,
+  identify: IdentifierAction,
+  getData: DataRetrievalActionProvider,
+  requireData: DataRequiredAction,
+  formProvider: TraderAgentRequiredInformationFormProvider,
+  val controllerComponents: MessagesControllerComponents,
+  view: TraderAgentRequiredInformationView
+)(implicit ec: ExecutionContext)
+    extends FrontendBaseController
+    with I18nSupport {
+
+  val form           = formProvider()
+  private val logger = play.api.Logger(getClass)
+
+  def onPageLoad(draftId: DraftId): Action[AnyContent] =
+    (identify andThen getData(draftId) andThen requireData) {
+      implicit request =>
+        logger.info("TraderAgentRequiredInformationController onPageLoad")
+        val preparedForm = request.userAnswers.get(TraderAgentRequiredInformationPage) match {
+          case None        => form
+          case Some(value) =>
+            form.fill(value)
+        }
```

Comment:
There's a handy helper you can use, so instead of:

`request.userAnswers.get(TraderAgentRequiredInformationPage)`

You can write:

`TraderAgentRequiredInformationPage.get()`